enhancements/operator-conditions.md

```diff
@@ -0,0 +1,188 @@
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The goal of this enhancement is to allow operators managed by the [Operator Lifecycle Manager (OLM)](https://github.com/operator-framework/operator-lifecycle-manager) operator to communicate conditions that OLM would otherwise be unable to infer. In particular, these conditions will be used by OLM to prevent irreversible actions (such as upgrades) when the operator is in a state that would be problematic.
+
+## Motivation
+
+As part of its role in managing the lifecycle of an operator, OLM infers the state of an operator from the state of Kubernetes resources that define the operator. While this sanity-check provides some level of assurance that an operator is in a given state, there are many complex states specific to each operator that OLM is unable to infer on its own. OLM could better handle the lifecycle of operators by providing them the means to communicate complex states. In particular, there are a number of benefits if an operator managed by OLM could communicate when it should not be upgraded. Consider the following scenario:
+
+>An operator installed with automatic updates is currently performing a migration that could result in data loss if OLM upgrades the operator. While this migration process is running, a new version of the operator becomes available. Since the operator was installed with automatic updates, OLM upgrades the operator without allowing the migration process to finish, resulting in unrecoverable data loss. Had OLM given the operator the ability to communicate that it should not be upgraded, no data loss would have occurred.
+
+As shown in the example above, by allowing operators to report complex condition and influence OLM's management, the overall user experience could be improved for both operator authors and OLM users.
+
+### Goals
+
+- Provide operators with the means to explicitly communicate conditions that influence OLM behavior.
+- Allow the list of "OLM Supported Conditions" to grow in the future.
+- Update OLM to support the "Upgrade Readiness" condition defined above.
+- Allow operators to communicate additional information to admins without influencing OLM behavior.
+- Preserve existing behavior if the operator does not take advantage of this feature.
+
+### Non-Goals
+
+- Provide operators with the means to cancel an upgrade that is in process.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an operator author, I want operators managed by OLM to be able to communicate complex states that OLM cannot infer on its own.
+
+#### Story 2
+
+As an operator author, I want operators managed by OLM to be able to communicate whether or not they are ready to be upgraded.
+
+## Design Details
+
+### Condition CRD
+
+OLM will create a channel for operators to communicate complex conditions by introducing the `Condition` [CustomResourceDefinition](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/).
```

Comment:
@bparees wht are your thoughts on the name of this resource? It seems much to generic to me but wanted a second opinion.


/cc @awgreene

Reply:
what is the group name?  Generic top level resource names can be fine if the full group name provides sufficient context.

Reply:
We discussed this a bit [here](https://github.com/operator-framework/enhancements/pull/42#discussion_r477314841) - full name would be the same as all OLM CRDs: `operators.coreos.com`